contracts/UsulVetoGuard.sol

```diff
@@ -0,0 +1,147 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./interfaces/IUsulVetoGuard.sol";
+import "./interfaces/IVetoVoting.sol";
+import "./interfaces/IOZLinearVoting.sol";
+import "./interfaces/IUsul.sol";
+import "./TransactionHasher.sol";
+import "./FractalBaseGuard.sol";
+import "@gnosis.pm/zodiac/contracts/factory/FactoryFriendly.sol";
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+/// @notice A guard contract that prevents transactions that have been vetoed from being executed a Gnosis Safe
+/// @notice through a Usul module with an attached OZLinearVoting strategy
+contract UsulVetoGuard is
+    IUsulVetoGuard,
+    TransactionHasher,
+    FactoryFriendly,
+    FractalBaseGuard
+{
+    IVetoVoting public vetoVoting;
+    IOZLinearVoting public ozLinearVoting;
+    IUsul public usul;
+    mapping(bytes32 => uint256) transactionQueuedBlock;
+
+    /// @notice Initialize function, will be triggered when a new proxy is deployed
+    /// @param initializeParams Parameters of initialization encoded
+    function setUp(bytes memory initializeParams) public override initializer {
+        __Ownable_init();
+        (
+            address _owner,
+            address _vetoVoting,
+            address _ozLinearVoting,
+            address _usul
+        ) = abi.decode(initializeParams, (address, address, address, address));
+
+        transferOwnership(_owner);
+        vetoVoting = IVetoVoting(_vetoVoting);
+        ozLinearVoting = IOZLinearVoting(_ozLinearVoting);
+        usul = IUsul(_usul);
+
+        emit UsulVetoGuardSetup(
+            msg.sender,
+            _owner,
+            _vetoVoting,
+            _ozLinearVoting,
+            _usul
+        );
+    }
+
+    /// @notice Queues a transaction for execution
+    /// @param proposalId The ID of the proposal to queue
+    function queueProposal(uint256 proposalId) external {
+        // If proposal is not yet timelocked, then finalize the strategy
+        if(usul.state(proposalId) == 0) ozLinearVoting.finalizeStrategy(proposalId);
+        
+        require(usul.state(proposalId) == 2, "Proposal must be timelocked before queuing");
+
+        // While loop is used since the Usul interface does not support getting the quantity of TX hashes
+        // stored within a given proposal. This loops through and gets the hash from each index until the call
+        // reverts, and then the function is exited
+        uint256 txIndex;
+        while (true) {
+            try usul.getTxHash(proposalId, txIndex) returns (bytes32 txHash) {
+                transactionQueuedBlock[txHash] = block.number;
+                txIndex++;
+            } catch {
+                require(txIndex > 0, "Invalid proposal ID");
+
+                emit ProposalQueued(msg.sender, proposalId);
+
+                return;
+            }
+        }
+    }
+
+    /// @notice This function is called by the Gnosis Safe to check if the transaction should be able to be executed
```

Comment:
Yeah @mudrila this guard is checked by the Usul mod before any transaction can be executed.